### PR TITLE
Disable XML Entity Expansions for unmarshalling

### PR DESCRIPTION
**Pull request by runeflobakk**
Ensuring JAXB marshalling (in particular `Unmarshaller`) uses a securely configured underlying XML parser, [according to recommendations by OWASP](https://cheatsheetseries.owasp.org/cheatsheets/XML_External_Entity_Prevention_Cheat_Sheet.html#jaxb-unmarshaller).

In addition, more generic support for customizing the `Marshaller` and `Unmarshaller`, and setting schema used for validation is retrofitted into this facility. This is used by the `BillionLaughsTest` to inject the infamous [Billion Laughs XXE attack](https://en.wikipedia.org/wiki/Billion_laughs_attack) header in order to test for the proper handling, i.e. not allowing a DOCTYPE header at all.

One additional unmarshal method is added: `unmarshal(Node, Class)`. This allows for separating the XML-parsing to a general DOM-representation, and then _afterwards_ unmarshal to a JAXB class from the already parsed DOM, instead of parsing the XML a second time.

### CODE REVIEW

File: jaxb/src/main/java/no/digipost/xml/bind/MarshallingCustomization.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        if (nextCustomization == NO_CUSTOMIZATION) {
+            return this;
+        } else if (this == NO_CUSTOMIZATION) {
+            return nextCustomization;
+        } else {
+            return of(
+                    this.marshallerCustomizer.andThen(nextCustomization.marshallerCustomizer),
+                    this.unmarshallerCustomizer.andThen(nextCustomization.unmarshallerCustomizer));
+        }
+    }
```
Comment (simenstoa) [inline jaxb/src/main/java/no/digipost/xml/bind/MarshallingCustomization.java:86]: Dette skjønte jeg ikke helt 😛 Hvis man sender inn `NO_CUSTOMIZATION` ignorerer vi det? Men hvis `NO_CUSTOMIZATION` allerede er satt (som er default) så endrer vi. Og hvis man sender inn flere etter det legger vi det til? Så man kan ha flere customizere?

File: jaxb/src/main/java/no/digipost/xml/bind/XmlUtils.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import static java.util.Objects.requireNonNull;
+
+final class XmlUtils {
+
+    public static Schema createSchema(Collection<String> schemaResourceNames) {
+        if (schemaResourceNames == null || schemaResourceNames.isEmpty()) {
+            throw new IllegalArgumentException("No sources given for creating Schema. (schemaResourceNames was " + schemaResourceNames + ")");
+        }
+        try {
+            SAXParserFactory parserFactory = SAXParserFactory.newInstance();
```
Comment (simenstoa) [inline jaxb/src/main/java/no/digipost/xml/bind/XmlUtils.java:46]: Hvorfor brukes ikke `SaxParserProvider.createSecuredParserFactory();` her?

File: jaxb/src/main/java/no/digipost/xml/bind/MarshallingCustomization.java (hunk 1 above)
Comment (simenstoa) [reply jaxb/src/main/java/no/digipost/xml/bind/MarshallingCustomization.java:86]: Det er litt ekstra greier for å sørge for å ikke utføre `NO_CUSTOMIZATION`-customizeren flere ganger enn nødvending?

Comment (runeflobakk) [reply jaxb/src/main/java/no/digipost/xml/bind/MarshallingCustomization.java:86]: Ja. Litt fancy-pants, men siden customizing gjøres _per_ marshalling/unmarshalling tenker jeg det var greit å unngå et "tomt" metodekall om mulig.

File: jaxb/src/main/java/no/digipost/xml/bind/XmlUtils.java (hunk 2 above)
Comment (runeflobakk) [reply jaxb/src/main/java/no/digipost/xml/bind/XmlUtils.java:46]: Den kan den nok! Dette ble nok tatt ut i en egen util før `SaxParserProvider` var tilgjengelig. Will fix!

**Review by simenstoa — APPROVED** [review]
👏

**Pull request merged**